R/derive_last_dose.R

```diff
@@ -82,40 +95,39 @@
 #'     head(ex_single, 100),
 #'     filter_ex = (EXDOSE > 0 | (EXDOSE == 0 & grepl("PLACEBO", EXTRT))) &
 #'       nchar(EXENDTC) >= 10,
+#'     ex_keep_vars = vars(EXSTDTC, EXENDTC, EXDOSE, EXTRT, EXSEQ, VISIT),
 #'     dose_start = EXSTDTC,
 #'     dose_end = EXENDTC,
 #'     analysis_date = AESTDTC,
 #'     dataset_seq_var = AESEQ,
-#'     new_var = LDOSEDTM,
-#'     output_datetime = TRUE,
 #'     check_dates_only = FALSE,
-#'     traceability_vars = dplyr::vars(LDOSEDOM = "EX", LDOSESEQ = EXSEQ, LDOSEVAR = "EXSTDTC")
+#'     traceability_vars = dplyr::vars(LDOSEDOM = "EX", LDOSESEQ = EXSEQ, LDOSEVAR = "EXENDTC")
 #'   ) %>%
-#'   select(STUDYID, USUBJID, AESEQ, AESTDTC, LDOSEDTM, LDOSEDOM, LDOSESEQ, LDOSEVAR)
+#'   select(STUDYID, USUBJID, AESEQ, AESTDTC, EXDOSE, EXTRT, EXENDTC, LDOSEDOM, LDOSESEQ, LDOSEVAR)
 #'
 derive_last_dose <- function(dataset,
                              dataset_ex,
                              filter_ex = NULL,
                              by_vars = vars(STUDYID, USUBJID),
+                             dose_id = vars(),
+                             ex_keep_vars = NULL,
```

Comment:
I think this question was mentioned in the last Monday meeting but I do not remember a decision.

According to the current programming strategy the parameter should be named `new_vars`. If we do not want to use this name, we should update the programming strategy and add a rule in which cases other names than `new_vars` should be used.

Reply:
I originally named it `ex_keep_vars` because I'm not sure if it's confusing `new_var` are source variables kept from EX dataset. However if it aligns with the programming strategy better maybe we can update to `new_vars` since the parameter description has explanations about where those variables should come from.

Reply:
Updated `ex_keep_vars` to `new_vars`.